File: app/src/main/java/com/princeakash/projectified/HomeAdapter.java
package com.princeakash.projectified;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class HomeAdapter extends RecyclerView.Adapter<HomeAdapter.HomeViewHolder> {

    private ArrayList<HomeItem> list;
    private Context context;
    private HomeListener homeListener;

    public HomeAdapter(ArrayList<HomeItem> list, Context context, HomeListener homeListener) {
        this.list = list;
        this.context = context;
        this.homeListener = homeListener;
    }

    @NonNull
    @Override
    public HomeViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.card_home_job, parent, false);
        return new HomeViewHolder(view, homeListener);
    }

    @Override
    public void onBindViewHolder(@NonNull HomeViewHolder holder, int position) {
        holder.textView.setText(list.get(position).itemName);
        holder.imageView.setImageResource(list.get(position).itemDrawable);
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public static class HomeViewHolder extends RecyclerView.ViewHolder{
        ImageView imageView;
        TextView textView;
        HomeListener homeListener;

        public HomeViewHolder(@NonNull View itemView, HomeListener homeListener) {
            super(itemView);
            this.imageView = itemView.findViewById(R.id.image);
            this.textView = itemView.findViewById(R.id.description);
            this.homeListener = homeListener;
        }
    }

    public static class HomeItem{
        public String itemName;
        public int itemDrawable;

        public HomeItem(String itemName, int itemDrawable) {
            this.itemName = itemName;
            this.itemDrawable = itemDrawable;
        }
    }

    public interface HomeListener{

    }
}

File: app/src/main/java/com/princeakash/projectified/candidate/addApplication/view/HomeAdapter.kt
package com.princeakash.projectified.candidate.addApplication.view

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.princeakash.projectified.R
import com.princeakash.projectified.candidate.addApplication.view.HomeAdapter.HomeViewHolder
import java.util.*


class HomeAdapter(private val list: ArrayList<HomeItem>, private val context: Context, private val homeListener: HomeListener) : RecyclerView.Adapter<HomeViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HomeViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.card_home_job, parent, false)
        return HomeViewHolder(view, homeListener)
    }

    override fun onBindViewHolder(holder: HomeViewHolder, position: Int) {
        holder.textView.text = list[position].itemName
        holder.imageView.setImageResource(list[position].itemDrawable)
    }

    override fun getItemCount(): Int {
        return list.size
    }

    class HomeViewHolder(itemView: View, homeListener: HomeListener) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        var imageView: ImageView
        var textView: TextView
        var homeListener: HomeListener

        init {
            imageView = itemView.findViewById(R.id.image)
            textView = itemView.findViewById(R.id.description)
            this.homeListener = homeListener
            itemView.setOnClickListener(this)
        }

        override fun onClick(v: View?) {
            homeListener.onDomainClick(adapterPosition)
        }
    }

    class HomeItem(var itemName: String, var itemDrawable: Int, var domainArg: String)

    interface HomeListener{
        fun onDomainClick(position:Int)
    }

    companion object{
        const val ANDROID = "android"
        const val WEB = "web"
        const val ML = "ml"
        const val AI = "ai"
        const val VIDEO = "video"
        const val UIUX = "uiux"
        const val CONTENT = "content"
        const val OTHERS = "others"
    }
}